shared/src/main/scala/mlscript/NewParser.scala

```diff
@@ -630,16 +630,34 @@ abstract class NewParser(origin: Origin, tokens: Ls[Stroken -> Loc], newDefs: Bo
   
   private def letBindings(genQuote: Bool)(implicit et: ExpectThen, fe: FoundErr, l: Line): IfBody \/ Term = {
     val bs = bindings(Nil)
-    val body = yeetSpaces match {
-      case (KEYWORD("in") | SEMI, _) :: _ =>
-        consume
-        exprOrIf(0)(et, fe, implicitly)
-      case (NEWLINE, _) :: _ =>
-        consume
-        exprOrIf(0)(et, fe, implicitly)
-      case _ =>
-        R(UnitLit(true).withLoc(curLoc.map(_.left)))
-    }
+    val body =
+      if (genQuote) yeetSpaces match {
+        case (QUOTE, l1) :: (KEYWORD("in"), l2) :: _ =>
+          consume
+          consume
+          yeetSpaces match {
+            case (NEWLINE, _) :: _ =>
+              consume
+              exprOrIf(0)(et, fe, implicitly)
+            case _ => exprOrIf(0)(et, fe, implicitly)
+          }
+        case (tk, loc) :: _ =>
+          err(msg"Expected '`in'; found ${tk.describe} instead" -> S(loc) :: Nil)
+          R(errExpr)
+        case Nil =>
+          err(msg"Expected '`in'; found end of input instead" -> lastLoc :: Nil)
+          R(errExpr)
```

Comment:
These error cases should be tested.

Reply:
I've added test cases in `shared/src/test/diff/qq/Weird.mls`:
```ts
:pe
:e
`let x = `1 `if
//│ ╔══[PARSE ERROR] This quote syntax is not supported yet
//│ ║  l.87: 	`let x = `1 `if
//│ ╙──      	         ^^
//│ ╔══[PARSE ERROR] Expected '`in'; found 'if' keyword instead
//│ ║  l.87: 	`let x = `1 `if
//│ ╙──      	             ^^
//│ ╔══[PARSE ERROR] Expected end of input; found 'if' keyword instead
//│ ║  l.87: 	`let x = `1 `if
//│ ╙──      	             ^^
//│ ╔══[ERROR] Type mismatch in unquote:
//│ ║  l.87: 	`let x = `1 `if
//│ ║        	               ^
//│ ╙── undefined literal of type `()` is not an instance of type `Code`
//│ Code[error, nothing]

:pe
:e
`let x = `1 `in
//│ ╔══[PARSE ERROR] Unexpected end of input; an expression was expected here
//│ ║  l.105: 	`let x = `1 `in
//│ ╙──       	               ^
//│ ╔══[ERROR] Type mismatch in unquote:
//│ ║  l.105: 	`let x = `1 `in
//│ ║         	               ^
//│ ╙── undefined literal of type `()` is not an instance of type `Code`
//│ Code[error, nothing]

```